### DISCUSSION TITLE
Interface to graph drawing libraries, or examples in documentation?

### DISCUSSION THREAD

**Discussion by HeWeMel** [HeWeMel/nographs#17] [Proposed features that need discussions]
NoGraphs could provide functionality to connect an implicit graph with the most common graph drawing libraries.

Should this be done? Why / why not? Alternatives?

**Comment by HeWeMel** [comment]
Pro: In easy cases, a (part of a) graph or the results of a graph traversal could be easily visualized without writing application code

**Comment by HeWeMel** [comment]
Contra: With NoGraphs, a graph is in the domain of the application, not the library.

**Comment by HeWeMel** [comment]
Contra: The code between a graph and such a library might be mostly application specific (styling, but also much more). This reduces the benefits of a standard interface.

**Comment by HeWeMel** [comment]
Contra: The connection of a graph with a drawing library is offered by these libraries, and they are good in this

**Comment by HeWeMel** [comment]
Contra: Library bloat

**Reply by leshabirukov** [reply]
I'll admit here nasty technical issue: `graphviz`, most useful backend, is not `pip install`able.

**Comment by HeWeMel** [comment]
Alternative: A documentation section with examples

**Reply by leshabirukov** [reply]
Well, it is related to my experiments, so, what can I say? At this point, I don't see strong reasons to change nog behaviour, or add core functionality. One can visulize some graph with little effort, at least, on simple cases. If users will repeatedly ask for something, that will be a reason to think.  
What remarkable challenges I had? Consider this: I want to collapse some edges, while keeping info about both nodes and the edge with composite node. But such a task is not typical, it is solvable, and mostly relates to my monstrous @'decorator stacks.
I think making more examples with other backends then `graphviz` will do the work. Preparing example whith `pyvis`.

**Reply by HeWeMel** [reply]
Yes, pyvis looks promising, too. And I found this article: [towardsdatascience](https://towardsdatascience.com/visualizing-networks-in-python-d70f4cbeb259).

(Currently, I have some other topics at work, but for next year I plan to extend the documentation by examples for getting, storing, exporting and drawing graphs. Thus, I am interested in experiences from connecting other libraries.)
